clusterloader2/pkg/measurement/common/slos/pod_startup_latency.go

```diff
@@ -140,42 +181,36 @@ func (p *podStartupLatencyMeasurement) gather(c clientset.Interface, identifier
 		return nil, err
 	}
 
-	podStartupLatency := p.podStartupEntries.CalculateTransitionsLatency(map[string]measurementutil.Transition{
-		"create_to_schedule": {
-			From: createPhase,
-			To:   schedulePhase,
-		},
-		"schedule_to_run": {
-			From: schedulePhase,
-			To:   runPhase,
-		},
-		"run_to_watch": {
-			From: runPhase,
-			To:   watchPhase,
+	checks := []podStartupLatencyCheck{
+		{
+			namePrefix: "",
+			filter:     measurementutil.MatchAll,
 		},
-		"schedule_to_watch": {
-			From: schedulePhase,
-			To:   watchPhase,
+		{
+			namePrefix: "Stateless",
+			filter:     p.podMetadata.FilterStateless(),
 		},
-		"pod_startup": {
-			From:      createPhase,
-			To:        watchPhase,
-			Threshold: p.threshold,
-		},
-	})
+	}
 
+	var summaries []measurement.Summary
 	var err error
-	if slosErr := podStartupLatency["pod_startup"].VerifyThreshold(p.threshold); slosErr != nil {
-		err = errors.NewMetricViolationError("pod startup", slosErr.Error())
-		klog.Errorf("%s: %v", p, err)
-	}
+	for _, check := range checks {
+		transitions := podStartupTransitionsWithThreshold(p.threshold)
+		podStartupLatency := p.podStartupEntries.CalculateTransitionsLatency(transitions, check.filter)
+
+		if slosErr := podStartupLatency["pod_startup"].VerifyThreshold(p.threshold); slosErr != nil {
```

Comment:
I think that stateful pods should use other threshold (and this validation should be disabled by default everywhere to make it possible to rollout it gradually)

Reply:
Note that in this PR I'm not really executing the stateful pods at all.
What I'm doing is:
- I'm leaving the existing "PodStartupLatency" as it was (includes all pods)
- I'm creating a new "StatelessPodsStartupLatency" (to include only stateless)

They both still use the same threshold at this point.


Once this soaks for a moment, we will create a dedicated StatefulPodsStartupLatency (yes - with a different treshold) and then sunset the original "PodStartupLatency". But both of these are further steps. I first want to make it more generic and see what we have for stateless explicitly.

Does that make sense?

Reply:
The new StatelessPodsStartupLatency can still generate metric violation and fail the test, but given that this is a subset of podstartuplatency pods which have the same property, we should be fine (unless we have some super fast stateful pods that reduces p99 of pod startup latency which is rather unlikely)

Reply:
OK yeah - there is an assumption that stateful pods don't start super fast. They barely ever fit into 5s,  so the stateless one shouldn't be higher in any of our existing tests.

Do you suggest setting it artificially higher now? Or are you ok as is for now?
[Note that we use 5s now only for the latency phase, where all pods are stateless anyway :) ]